From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: IBM Codestream linux-2.6.16 october2005, patch base-06
Patch-Mainline: 2.6.17-rc1

[patch] kernel: BUG() warnings.

Use __builtin_trap instead of an inline assembly in the BUG() macro.
That way the compiler knows that BUG() won't return.
Acked-by: Hannes Reinecke <hare@suse.de>

---

 include/asm-s390/bug.h |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff -urpN linux-2.6/include/asm-s390/bug.h linux-2.6-patched/include/asm-s390/bug.h
--- linux-2.6/include/asm-s390/bug.h	2006-03-20 06:53:29.000000000 +0100
+++ linux-2.6-patched/include/asm-s390/bug.h	2006-03-22 11:46:45.000000000 +0100
@@ -4,9 +4,10 @@
 #include <linux/kernel.h>
 
 #ifdef CONFIG_BUG
+
 #define BUG() do { \
-        printk("kernel BUG at %s:%d!\n", __FILE__, __LINE__); \
-        __asm__ __volatile__(".long 0"); \
+	printk("kernel BUG at %s:%d!\n", __FILE__, __LINE__); \
+	__builtin_trap(); \
 } while (0)
 
 #define HAVE_ARCH_BUG
